test/unit/test_gateway.c

```diff
@@ -592,6 +594,41 @@ TEST_CASE(prepare, non_leader, NULL)
 	return MUNIT_OK;
 }
 
+/* Try to prepare a string containing more than one statement. */
+TEST_CASE(prepare, nonempty_tail, NULL)
+{
+	struct prepare_fixture *f = data;
+	(void)params;
+	f->request.db_id = 0;
+	f->request.sql = "CREATE TABLE test (n INT); SELECT * FROM test";
+	CLUSTER_ELECT(0);
+	ENCODE(&f->request, prepare);
+	HANDLE(PREPARE);
+	WAIT;
+	ASSERT_CALLBACK(0, FAILURE);
+	ASSERT_FAILURE(SQLITE_ERROR, "nonempty statement tail");
+	return MUNIT_OK;
+}
+
+/* Try to prepare a string containing more than one statement, successfully. */
+TEST_CASE(prepare, nonempty_tail_v1, NULL)
+{
+	struct prepare_fixture *f = data;
+	struct response_stmt_with_offset response = {0};
+	(void)params;
+	f->request.db_id = 0;
+	f->request.sql = "CREATE TABLE test (n INT); SELECT * FROM test";
+	CLUSTER_ELECT(0);
+	ENCODE(&f->request, prepare);
+	HANDLE_SCHEMA_STATUS(DQLITE_REQUEST_PREPARE, 1, 0);
+	WAIT;
+	ASSERT_CALLBACK(0, STMT_WITH_OFFSET);
+	DECODE(&response, stmt_with_offset);
+	munit_assert_int(response.id, ==, 0);
+	munit_assert_uint64(response.offset, ==, 26);
```

Comment:
I'm being nitty here, but it would be nice if we used the offset returned by dqlite to index into the original SQL string and try to successfully prepare the next statement.